services/cd-service/pkg/repository/repository.go

```diff
@@ -362,7 +384,9 @@ type PushActionCallbackFunc func(git.PushOptions, *repository) PushActionFunc
 // DefaultPushActionCallback is public for testing reasons only.
 func DefaultPushActionCallback(pushOptions git.PushOptions, r *repository) PushActionFunc {
 	return func() error {
-		return r.remote.Push([]string{fmt.Sprintf("refs/heads/%s:refs/heads/%s", r.config.Branch, r.config.Branch)}, &pushOptions)
+		return r.useRemote(context.Background(), func(remote *git.Remote) error {
```

Comment:
What exactly is the difference here?

Reply:
The difference is that `useRemote` opens a new connection every time. Also it ensures that the connection is closed at the end and, most importantly, it is guarateed to return after `NetworkTimeout` so that calls don't stack up.